src/robotFactory.js

```diff
@@ -1,10 +1,75 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = coords;
+    this.coords.x = coords.x || 0;
+    this.coords.y = coords.y || 0;
```

Comment:
You can group the coords properties into object, like this.

```suggestion
   
   this.coords = coords {
     x: coords.x || 0;
     y: coords.y || 0;
    }
```

Reply:
Yes, I agree) Thanks!)